mypy/checker.py

```diff
@@ -2756,3 +2749,51 @@ def is_valid_inferred_type_component(typ: Type) -> bool:
             if not is_valid_inferred_type_component(item):
                 return False
     return True
+
+
+class Scope:
+    # We keep two stacks combined, to maintain the relative order
+    stack = None  # type: List[Union[Type, FuncItem, MypyFile]]
+
+    def __init__(self, module: MypyFile) -> None:
+        self.stack = [module]
+
+    def push_function(self, item: FuncItem) -> None:
+        self.stack.append(item)
+
+    def pop_function(self) -> None:
+        assert isinstance(self.stack[-1], Type)
```

Comment:
This seems oddly mismatched with the argument type of push_function(). And since the assert never failed in the tests I wonder if this code is exercised at all?